### DISCUSSION TITLE
Which ROM AOSP 12 for ARM 64 A Only device ?

### DISCUSSION THREAD

**Discussion by ED4free** [phhusson/treble_experimentations#2534] [General]
Hello, 
Thanks again @phhusson for that useful work !
My device (Blackview BV4900 Pro, Mediatek MT6762D)  sold with Android 12 is Treble compliant, VNDK 31.0, system partition A only, ARM64.
I don't see any A Only ROM for AOSP12. So which ROM must I take ?
Thierry

**Comment by JonnyTech** [comment]
A-only images were discontinued after Android 11 - in fact, Google stopped them beforehand and phhusson kept them going for longer (thanks phh!), but alas no more. Are you sure that your device is really a-only?

**Comment by phhusson** [comment]
Your device uses AB GSI images.

**Comment by ED4free** [comment]
Thank you both for your answer @JonnyTech and @phhusson 
The application Treble Chek tell it is A Only. Cf screenshot.  
![Screenshot_20220726-175936](https://user-images.githubusercontent.com/32482942/230796365-af2452bb-0a09-452c-97b4-dbbbf24a3d2c.png).
I continue to test with AB ROM, but for the moment  : 
The "fastboot flash system XXX" says "Invalid sparse file format at header magic", but after the smartphone don't boot.
In fastbootd mode there is written "Secure boot - yes", perhaps it's why it don't boot.

**Comment by JonnyTech** [comment]
We wish that people would stop using Treble Check, please downvote or leave a negative review, then use Treble Info instead: https://play.google.com/store/apps/details?id=tk.hack5.treblecheck

> The "fastboot flash system XXX" says "Invalid sparse file format at header magic"

That message is not an error message.

Did you flash  `vbmeta` to disable AVB?

**Reply by ED4free** [reply]
Thanks for this advice.
Yes,  I tried both, with and without "fastboot --disable-verity --disable-verification flash vbmeta vbmeta.img"
I downloaded vbmeta.img on this page : https://forum.xda-developers.com/t/guide-phh-gsi-flash-aosp-12.4328891/
After I go in recovery mode and make "wipe cache partition" and "wipe data" 
The device try to boot, wait 10 second, stop and try again,  it make loop.

**Comment by JonnyTech** [comment]
Try https://dl.google.com/developers/android/qt/images/gsi/vbmeta.img

Did you format data parition from stock recovery and not custom recovery such as TWRP?

Which GSI did you flash? Please provide a link.

**Comment by ED4free** [comment]
I tried with vbmeta.img you provided me. The result is a little different. The device don't boot, it is blocked. With the other one it was trying every 30 seconds.

Yes I format data partition from stock recovery, not TWRP.

I took GSI from here : https://github.com/phhusson/treble_experimentations/releases/tag/v416
I tried with : 
AOSP-12-v416-system-squeak-arm64-ab-gogapps.img
AOSP12-v416-system-squeak-arm64-ab-vanilla.img
AOSP-12-V416-system-squeak-arm64-ab-vndklite-gogapps.img

**Comment by JonnyTech** [comment]
Try stock `vbmeta.img`, with and without the disable options (you may need to reflash stock `boot.img` first)

**Reply by ED4free** [reply]
Done.
I flashed the stock vbmeta.img , with and without options.  Same result.
Then I flashed stock boot.img. Same result.
Then I flashed again the stock vbmeta.img. Same result.

In [this post ](https://forum.xda-developers.com/t/solved-gsi-on-bv-4900-pro-cannot-load-android-system-your-data-may-be-corrupt.4531187/) he says : 

===

I just realized / remembered that this phone uses super.img instead of system.img, maybe I need to repack the system.img into a super.img, following [this procedure](https://forum.xda-developers.com/t/solved-which-partition-to-flash-to-install-lineageos-project-treble-from-twrp-on-blackview-bv-4900-pro.4453955/#post-87029001)?

===

Do you think it could be the same problem ?
I look at my stock ROM, I confirm there is no system.img but super.img
So it should be that.
Thanks

**Reply by phhusson** [reply]
How did you flash system image?

**Reply by ED4free** [reply]
fastboot erase system
fastboot flash system AOSP-12-v416-system-squeak-arm64-ab-gogapps.img
then wipe cache and data

**Reply by ED4free** [reply]
Notice that in my stock ROM I have a file super.img but no file system.img
Don't you think it is why it does not work ?

**Reply by phhusson** [reply]
No that's not the reason, it's just that system is inside super.img, but when you did `fastboot flash system` it did modified just the system part of super.img

Could you try with https://github.com/TrebleDroid/treble_experimentations/releases/download/ci-20230330/system-td-arm64-ab-vanilla.img.xz ?

**Reply by ED4free** [reply]
It works !
I didn't flash vbmeta.
It is strange : there was no process of configuring android (language, wifi, accet conditions, etc...)

**Reply by ED4free** [reply]
It is Android 13. Already rooted.
I guess that now you know what was the problem.
So what ROM do you recommend ?
If you remember our mail exchanges 2 years ago, it is not for personal use. I install on it a small Debian server and then software and content for education in Africa (this model will be for Tchad).
I don't mind if it is Android 10,11,12,13
I don't need any application.
I only need to install Linux Deploy, and to start a shell at boot and for that Magisk is very easy (in Andoid 10 I used init.rc, but I noticed that after it was no more possible because /system/bin was reinit a t boot). 
Thanks a lot.

**Reply by phhusson** [reply]
>  It is strange : there was no process of configuring android (language, wifi, accet conditions, etc...)
Yes that's expected, it's a "vanilla" variant, so it includes only bare minimum apps. No Google apps or anything. Not even an actual web browser.

>  If you remember our mail exchanges 2 years ago, it is not for personal use. I install on it a small Debian server and then software and content for education in Africa (this model will be for Tchad).

Yup, (I Just didn't realize it initially)

>  I guess that now you know what was the problem.

Yes it's https://github.com/TrebleDroid/device_phh_treble/commit/812b0e8bff7f4f40ca3c168c7c220c290f7842fa + https://github.com/TrebleDroid/device_phh_treble/commit/918cfbff1deb77cc7c5b921a377c208d1f433b6f + https://github.com/TrebleDroid/device_phh_treble/commit/7ad38023e53279dff478718d2e74a481a015bcda

Those patches could be integrated in Android 12

> (in Andoid 10 I used init.rc, but I noticed that after it was no more possible because /system/bin was reinit a t boot).

Disable magisk temporarily or put it into "core mode" and you'll be able to write to /system

>  I don't mind if it is Android 10,11,12,13
FWIW Treble doesn't allow downgrade. So your original OEM ROM is Android 12, you can't go lower than Android 12.

> I only need to install Linux Deploy, and to start a shell at boot and for that Magisk is very easy 
I don't think you actually need magisk for that, the integrated root should be enough. But use whatever suits you.

**Reply by ED4free** [reply]
Thanks a lot ! 
I'll test several configurations and choose the best one. Perhaps Vanilla will be fine because it is very light.
NB : 15 mini educational server based on Smartphone with PHH ROM are already running in Madagascar schools.  Next countries : Tchad, Mauritania, Gabon, Senegal. It will be demonstrate at next exhibition "eLearning Africa" end May in Dakar.

**Reply by phhusson** [reply]
Cool. thanks for the update

**Comment by JonnyTech** [comment]
I suggest asking in the Telegram support group: https://t.me/phhtreble
